src/python_testing/TC_OpstateCommon.py

```diff
@@ -0,0 +1,1191 @@
+        if self.is_ci:
+            self.write_to_app_pipe(command)
+        else:
+            input("Press Enter when done.\n")
+
+    async def send_cmd(self, endpoint, cmd, timedRequestTimeoutMs=None):
+        logging.info("\t##### Command %s" % cmd)
+
+        try:
+            return await self.send_single_cmd(cmd=cmd,
+                                              endpoint=endpoint,
+                                              timedRequestTimeoutMs=timedRequestTimeoutMs)
+
+        except InteractionModelError as e:
+            asserts.fail("Unexpected error returned: %s" % e.status)
+
+    async def send_cmd_expect_response(self, endpoint, cmd, expected_response, timedRequestTimeoutMs=None):
+        ret = await self.send_cmd(endpoint=endpoint,
+                                  cmd=cmd,
+                                  timedRequestTimeoutMs=timedRequestTimeoutMs)
+
+        asserts.assert_equal(ret.commandResponseState.errorStateID,
+                             expected_response,
+                             "Current value not according expected. Expected %d, received %d!" %
+                             (expected_response, ret.commandResponseState.errorStateID))
+
+    async def read_expect_success(self, endpoint, attribute):
+        logging.info("\t##### Read %s" % attribute)
+        attr_value = await self.read_single_attribute_check_success(endpoint=endpoint,
+                                                                    cluster=self.test_info.cluster,
+                                                                    attribute=attribute)
+        logging.info("\t## %s: %s" % (attribute, attr_value))
+
+        return attr_value
+
+    async def read_and_expect_value(self, endpoint, attribute, expected_value):
+        attr_value = await self.read_expect_success(
+            endpoint=endpoint,
+            attribute=attribute)
+
+        asserts.assert_equal(attr_value, expected_value,
+                             "Current value not according expected. Expected %d, received %d!" %
+                             (expected_value, attr_value))
+
+    async def read_and_expect_property_value(self, endpoint, attribute, attr_property, expected_value):
+        attr_value = await self.read_expect_success(
+            endpoint=endpoint,
+            attribute=attribute)
+        attr_value = getattr(attr_value, attr_property)
+
+        asserts.assert_equal(attr_value, expected_value,
+                             "Current value not according expected. Expected %d, received %d!" %
+                             (expected_value, attr_value))
+
+    async def read_and_expect_array_contains(self, endpoint, attribute, expected_contains):
+        attr_value = await self.read_expect_success(
+            endpoint=endpoint,
+            attribute=attribute)
+        attr_value.sort()
+        expected_contains.sort()
+
+        logging.info("\t## Current value: [%s]" % ','.join(map(str, attr_value)))
```

Comment:
There is not need to self-format on comma-separated. The %s on a Python list will log the correct thing.

Reply:
done